### PR TITLE
draft: add query helpers

### PR DESCRIPTION
**Pull request by jeswr**
This is a WIP PR adding query helper functionality to the core of SolidLabLib.js

### CODE REVIEW

**Comment by rubensworks** [comment]
Do you want a review at this stage @jeswr?

File: packages/core/lib/SolidUtilContext.ts [hunk 1]
```diff
@@ -1,25 +1,55 @@
...
+ */
+export type SolidUtilContext =
+  IQueryAlgebraContext<Algebra.Operation, SparqlResultSupport, QueryAlgebraContext & QuerySourceContext<string>> &
+  IQueryStringContext<SparqlResultSupport, QueryStringContext & QuerySourceContext<string>> & {
+  /**
+   * An authenticated Solid session.
+   */
+    session?: Session;
+  };
+
```
Comment (jeswr) [inline packages/core/lib/SolidUtilContext.ts:52]: @rubensworks not after a full review yet as there is still more to port. I just want to know if you are on board with the typing changes in this file.

Comment (rubensworks) [reply packages/core/lib/SolidUtilContext.ts:52]: Could you clarify a bit what the purpose of this PR is exactly?

I see a lot of query-related helpers being added in here.
Are you adding them just for the sake of having them? Or are they in function of new Solid-related packages that you'll be creating?
The helpers that I had added before are just here to enable the `getIdp()` functionality, but I have the feeling that the new helpers have no immediate purpose (yet?).

I just want the project to remain focused on its main goal, and avoid adding logic that would increase maintenance effort.
Unless of course we want to broaden the scope of this project, and also add a generic query toolkit.
Also looking at @RubenVerborgh for this.

Comment (jeswr) [reply packages/core/lib/SolidUtilContext.ts:52]: >  Are you adding them just for the sake of having them? Or are they in function of new Solid-related packages that you'll be creating?

These are functions that I've found myself re-writing for several use-cases (that are indeed query related) where the query engine is passed around in a config. These particular implementations are coming from a utility package in a query PoC that I'm aiming to make public this week. With that said, it probably does make sense to omit functions with hard-coded terms such as `queryLabel`.

> I just want the project to remain focused on its main goal ... unless of course we want to broaden the scope of this project, and also add a generic query toolkit.

I'm not entirely clear on what the full scope / goal of this project is indeed so it's is hard to judge exactly how much to port over here without that context.

Comment (jeswr) [reply packages/core/lib/SolidUtilContext.ts:52]: Quite happy to move these to an `@solidlab/query` package if that is more suitable; or close this PR if these types of functions don't belong here. @rubenverborgh do you have a position on whether or not this kind of thing is in scope?

**Comment by rubensworks** [comment]
@jeswr and I just discussed this. Since these helpers are mostly Solid-independent, and exist at RDF/JS query-engine-level, we're going to create a dedicated RDF/JS query-engine helpers package, which will exist under the RDF/JS namespace (after discussion in the RDF/JS CG).

**Pull request closed without merging**